src/main/java/io/stargate/sgv2/jsonapi/service/resolver/CommandResolver.java

```diff
@@ -213,4 +216,24 @@ private void getIndexUsageTags(DBLogicalExpression dbLogicalExpression, IndexUsa
       getIndexUsageTags(subDBLogicalExpression, indexUsage);
     }
   }
+
+  /**
+   * Validate the name of a schema object (e.g., keyspace, table, collection, or index).
+   *
+   * @param name The name to validate.
+   * @param namingRule The naming rule to apply.
+   * @throws SchemaException if the name is invalid.
+   */
+  default void validateSchemaName(String name, SchemaObjectNamingRule namingRule) {
+    if (!namingRule.apply(name)) {
+      throw SchemaException.Code.UNSUPPORTED_SCHEMA_NAME.get(
+          Map.of(
+              "schemaType",
+              namingRule.name(),
+              "maxNameLength",
+              String.valueOf(SchemaObjectNamingRule.MAX_NAME_LENGTH),
```

Comment:
using SchemaObjectNamingRule.MAX_NAME_LENGTH feels wrong, why is it static and not on the object ?